src/main/java/cz/cvut/kbss/termit/websocket/handler/WebSocketExceptionHandler.java

```diff
@@ -92,8 +97,61 @@ private static ErrorInfo errorInfo(Message<?> message, TermItException e) {
                                                        e.getParameters());
     }
 
+    /**
+     * Tries to match method on this object that matches signature with params
+     *
+     * @return true when a method was found and called, false otherwise
+     * @throws IllegalArgumentException never
+     */
+    public boolean delegate(Message<?> message, Throwable throwable) {
+        try {
+            return delegateInternal(message, throwable.getCause());
+        } catch (InvocationTargetException invEx) {
+            LOG.error("Exception thrown during exception handler invocation", invEx);
+        } catch (IllegalAccessException unexpected) {
+            // is checked by delegate
+        }
+        return false;
+    }
+
+    /**
+     * Tries to match method on this object that matches signature with params
+     *
+     * @return true when a method was found and called, false otherwise
+     * @throws IllegalArgumentException never
+     */
+    private boolean delegateInternal(Message<?> message, Throwable throwable)
+            throws InvocationTargetException, IllegalAccessException {
+        if (throwable instanceof Exception exception) {
+            Method[] methods = this.getClass().getMethods();
+            for (final Method method : methods) {
+                if (!method.canAccess(this) || method.getName().equals("delegate") || method.getName().equals("delegateInternal")) {
+                    continue;
+                }
+                Class<?>[] params = method.getParameterTypes();
+                if (params.length != 2) {
+                    continue;
+                }
+                if (params[0].isAssignableFrom(message.getClass()) && params[1].isAssignableFrom(exception.getClass())) {
+                    // message, exception
+                    method.invoke(this, message, exception);
+                    return true;
+                } else if (params[0].isAssignableFrom(exception.getClass()) && params[1].isAssignableFrom(message.getClass())) {
+                    // exception, message
+                    method.invoke(this, exception, message);
+                    return true;
+                }
```

Comment:
This branch should not be necessary. If it is, let's rather refactor the code to have a consistent API in this class.